nilearn/image/image.py

```diff
@@ -364,16 +376,53 @@ def crop_img(img, rtol=1e-8, copy=True):
     if data.ndim == 4:
         passes_threshold = np.any(passes_threshold, axis=-1)
     coords = np.array(np.where(passes_threshold))
-    start = coords.min(axis=1)
-    end = coords.max(axis=1) + 1
+
+    # Sets full range if no data are found along the axis
+    if coords.shape[1] == 0:
+        start, end = [0, 0, 0], list(data.shape)
+    else:
+        start = coords.min(axis=1)
+        end = coords.max(axis=1) + 1
 
     # pad with one voxel to avoid resampling problems
-    start = np.maximum(start - 1, 0)
-    end = np.minimum(end + 1, data.shape[:3])
+    if pad:
+        start = np.maximum(start - 1, 0)
+        end = np.minimum(end + 1, data.shape[:3])
+
+    slices = [slice(s, e) for s, e in zip(start, end)][:3]
+    cropped_im = _crop_img_to(img, slices, copy=copy)
+    return cropped_im if not return_offset else (cropped_im, tuple(slices))
 
-    slices = [slice(s, e) for s, e in zip(start, end)]
 
-    return _crop_img_to(img, slices, copy=copy)
+def pad_img(array, *args):
```

Comment:
I would rather avoid the *args pattern and use "pad_sizes" as an explicit required argument:
```suggestion
def pad_img(array, pad_sizes):
```

Reply:
Also, if the input is an array, the function should be called "pad_array". But in this case, it should also probably be made private: "_pad_array"

Reply:
Agreed - I just inherited this from the previously written utility I copied over. :)